src/languageServerApi/languageServerApiManager.ts

```diff
@@ -1,94 +1,114 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license.
 
-import { commands, Event, Extension, extensions, Uri } from "vscode";
+import { commands, Disposable, Event, Extension, extensions, Uri } from "vscode";
 import { Commands } from "../commands";
+import { Context, ExtensionName } from "../constants";
 import { contextManager } from "../contextManager";
+import { Settings } from "../settings";
+import { syncHandler } from "../syncHandler";
 import { LanguageServerMode } from "./LanguageServerMode";
 
-class LanguageServerApiManager {
+class LanguageServerApiManager implements Disposable {
+    /**
+     * undefined means a legacy version language server
+     * null means the JDT.LS is not activated
+     */
     private serverMode: LanguageServerMode | null | undefined = null;
 
-    public async isStandardServerReady(): Promise<boolean> {
+    private extensionChangeListener: Disposable;
+
+    public async ready(): Promise<boolean> {
         await this.checkServerMode();
-        // undefined serverMode indicates an older version language server
-        if (this.serverMode === undefined) {
-            return true;
-        }
 
-        if (this.serverMode !== LanguageServerMode.Standard) {
+        if (this.serverMode === null || this.serverMode === LanguageServerMode.LightWeight) {
```

Comment:
what's the difference of this line? Or just clearer?

Reply:
In which case, will this.serverMode be LanguageServerMode.Hybrid? On startup?

Reply:
When both lightweight server and standard server are working -> mostly standard server is activating, since once it's activated, lightweight server will be killed.